src/main/scala/org/broadinstitute/hail/stats/eigSymD.scala

```diff
@@ -0,0 +1,139 @@
+package org.broadinstitute.hail.stats
+
+import breeze.linalg._
+import breeze.generic.UFunc
+import com.github.fommil.netlib.LAPACK.{getInstance => lapack}
+import org.netlib.util.intW
+
+/**
+  * Computes all eigenvalues (and optionally right eigenvectors) of the given
+  * real symmetric matrix X using dsyevd (Divide and Conquer):
+  * `http://www.netlib.org/lapack/explore-html/d2/d8a/group__double_s_yeigen_ga694ddc6e5527b6223748e3462013d867.html`
+  *
+  * Based on eigSym in breeze.linalg.eig but replaces dsyev with dsyevd for higher performance:
+  * http://www.netlib.org/lapack/lawnspdf/lawn183.pdf
+  */
+object eigSymD extends UFunc {
+  case class eigSymD[V, M](eigenvalues: V, eigenvectors: M)
+  type DenseeigSymD = eigSymD[DenseVector[Double], DenseMatrix[Double]]
+  implicit object eigSymD_DM_Impl extends Impl[DenseMatrix[Double], DenseeigSymD] {
+    def apply(X: DenseMatrix[Double]): DenseeigSymD = {
+      doeigSymD(X, rightEigenvectors = true) match {
+        case (ev, Some(rev)) => eigSymD(ev, rev)
+        case _ => throw new RuntimeException("Shouldn't be here!")
+      }
+
+    }
+  }
+
+  object justEigenvalues extends UFunc {
+    implicit object eigSymD_DM_Impl extends Impl[DenseMatrix[Double], DenseVector[Double]] {
+      def apply(X: DenseMatrix[Double]): DenseVector[Double] = {
+        doeigSymD(X, rightEigenvectors = false)._1
+      }
+    }
+
+  }
+
+  private def doeigSymD(X: Matrix[Double], rightEigenvectors: Boolean): (DenseVector[Double], Option[DenseMatrix[Double]]) = {
+    //requireNonEmptyMatrix(X)
+    //requireSymmetricMatrix(X)
```

Comment:
If you leave these in commented out, make a short comment saying why they're there (e.g. assume X is non-empty, symmetrix.  Check too expensive, disabling by default.)  In general, I'm okay with (amortized) O(1) checks staying in at this point.